stl/inc/coroutine

```diff
@@ -0,0 +1,168 @@
+// coroutine standard header (core)
+
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#pragma once
+#ifndef _COROUTINE_
+#define _COROUTINE_
+#include <yvals_core.h>
+#if _STL_COMPILER_PREPROCESSOR
+
+#ifdef _RESUMABLE_FUNCTIONS_SUPPORTED
+#pragma message("The contents of <coroutine> are not available with /await. Use <experimental/coroutine> instead.")

```

Comment:
Is it really the best advice to move from standard to experimental?

Reply:
If you're already using /await then you are likely already using experimental coroutines, but this may not be entirely clear. Perhaps something like:

```c++
#pragma message("The contents of <coroutine> are not available with /await.")
#pragma message("Remove /await for standard coroutines or use <experimental/coroutine>.")

```
I'll defer to the STL devs on their preferred wording, though.

Reply:
That wording sounds good to me.

Reply:
How about:
```
For standard coroutines, compile with C++20, do not specify /await, and include <coroutine>.
For TS coroutines, specify /await and include <experimental/coroutine>.
```
"TS" is probably too obscure, even expanded to "Technical Specification". "Legacy coroutines"?

Reply:
"non-standard" or just simply "otherwise"?

Reply:
I didn't want to say TS because /await is not exactly the coroutines TS. If we're going to say standard/non-standard than I prefer my first proposed wording.

Reply:
I could live with "non-standard" or "legacy" - anything that suggests the result is icky enough that it's not what you're looking for. We want to steer people away from "old" coroutines but without giving existing users the idea that they will be abadoned.

Reply:
"/await is incompatible with standard coroutines; remove it or use <experimental/coroutine>"?

Reply:
Legacy is probably a bit more useful of a term than 'non-standard' here. "Non-standard" could be just about anything, but this is specifically to maintain compatibility for existing /await users.

Is the STL using the name C++20 in messages? I want to avoid specifically naming the /std:c++latest flag unless necessary to avoid another update when we introduce the 20 switch.

Reply:
> Is the STL using the name C++20 in messages? 

Yes we are. Line 16, for example.

Reply:
How about:
```c++
#pragma message("The contents of <coroutine> are not available with /await.")
#pragma message("Remove /await for standard coroutines or use <experimental/coroutine> for legacy /await support.")
```
?

Reply:
That puts enough stink on `/await` to steer people away - I'm happy.